zerver/tests/test_outgoing_webhook_system.py

```diff
@@ -111,3 +111,58 @@ def test_mention_message_event_flow(self, mock_queue_json_publish):
         self.subscribe_to_stream(self.bot_profile.email, "Denmark")
         self.send_message(self.user_profile.email, "Denmark", Recipient.STREAM, self.content)
         self.assertTrue(mock_queue_json_publish.called)
+
+class PersonalMessageTriggersTest(ZulipTestCase):
+
+    def setUp(self):
+        # type: () -> None
+        user_profile = get_user_profile_by_email("othello@zulip.com")
+        self.bot_user = do_create_user(email="testvabs-bot@zulip.com",
+                                       password="test",
+                                       realm=get_realm_by_email_domain("zulip.com"),
+                                       full_name="The Test Bot",
+                                       short_name="bot",
+                                       bot_type=UserProfile.OUTGOING_WEBHOOK_BOT,
+                                       bot_owner=user_profile)
+
+    def test_trigger_on_stream_message(self):
+        # type: () -> None
+        sender_email = "othello@zulip.com"
+        recipients = "Denmark"
+        message_type = Recipient.STREAM
+        with mock.patch('zerver.lib.actions.queue_json_publish') as queue_json_publish:
+            self.send_message(sender_email, recipients, message_type)
+            self.assertFalse(queue_json_publish.called)
+
+    def test_trigger_on_private_message_by_bot(self):
+        # type: () -> None
+        sender_email = "testvabs-bot@zulip.com"
+        recipients = "othello@zulip.com"
+        message_type = Recipient.PERSONAL
+
+        with mock.patch('zerver.lib.actions.queue_json_publish') as queue_json_publish:
+            self.send_message(sender_email, recipients, message_type)
+            self.assertFalse(queue_json_publish.called)
+
+    @mock.patch('zerver.lib.actions.queue_json_publish')
+    def test_trigger_on_private_message_by_user(self, mock_queue_json_publish):
```

Comment:
I think we can add one for single/multiple users, multiple bots huddle case. In that we can check if multiple triggers are generated.